docs/plugins.md

```diff
@@ -136,25 +136,54 @@ function preprocess(text: string, options: object): string;
 
 Printers convert ASTs into a Prettier intermediate representation, also known as a Doc.
 
-The key must match the `astFormat` that the parser produces. The value contains an object with a `print` function and (optionally) an `embed` function.
+The key must match the `astFormat` that the parser produces. The value contains an object with a `print` function. All other properties (`embed`, `preprocess`, etc.) are optional.
 
 ```js
 export const printers = {
   "dance-ast": {
     print,
     embed,
+    preprocess,
     insertPragma,
+    canAttachComment,
+    isBlockComment,
+    printComment,
+    handleComments: {
+      ownLine,
+      endOfLine,
+      remaining,
+    },
   },
 };
 ```
 
-Printing is a recursive process of converting an AST node (represented by a path to that node) into a doc. The doc is constructed using the [builder commands](https://github.com/prettier/prettier/blob/master/commands.md):
+#### The printing process
+
+Prettier uses an intermediate representation, called a
+Doc, which Prettier then turns into a string (based on
+options like `printWidth`). A _printer_'s job is to take
+the AST generated by `parsers[<parser name>].parse` and
+return a Doc. A Doc is constructed using [builder
+commands](https://github.com/prettier/prettier/blob/master/commands.md):
```

Comment:
Can we have these paragraphs in one line?! Editors and github are handling that perfectly.. and it's much easier to change in the future, if we need to update the copy.

Reply:
I thought Jason did this intentionally to make it easier to add review comments in the middle of a paragraph. Before merging we'll reformat the file with `--prose-wrap always`.

Reply:
That's exactly right. I wrapped the text to make it easier to review and add diffs. prosewrap seems like a good solution :-)

Reply:
Awesome!